libcloudforensics/prompts.py

```diff
@@ -66,22 +69,26 @@ def text(self) -> str:
     """The text description of this PromptOption."""
     return self._text
 
-  def Disable(self) -> None:
-    """Disables this prompt."""
-    self._disabled = True
+  def Disable(self, reason: str) -> None:
+    """Disables this prompt.
+
+    Args:
+      reason (str): The reason for disabling this prompt.
+    """
+    self._disabled_reason = reason
 
   def IsDisabled(self) -> bool:
     """Returns True if this prompt is disabled, false otherwise.
 
     Returns:
       bool: True if this prompt is disabled, false otherwise.
     """
-    return self._disabled
+    return self._disabled_reason is not None
 
   def Select(self) -> None:
     """Selects this prompt, disabling dependent prompts."""
-    for option in self._disable_options:
-      option.Disable()
+    for option, reason in self._disable_options:
```

Comment:
`self._disable_options` is a list of Tuple, so I expect you're going to have a `TypeError` here.
You want to first iterate on that list, and then unpack the tuple into `option` and `reason`
```python
for disable in self._disable_options:
  for option, reason in disable:
    option.Disable(reason)
```

Reply:
Values of a list can be unpacked directly with the variables given in the for loop: 

```
>>> l = [(0, 'foo'), (1, 'bar')]
>>> for i, val in l:
...     print(i, val)
... 
0 foo
1 bar
>>> for i, val in enumerate(['a', 'b', 'c']):
...     print(i, val)
... 
0 a
1 b
2 c

```

If I try out the code snippet:

```
>>> for tup in l:
...     for i, val in tup:
...         print(i, val)
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
TypeError: cannot unpack non-iterable int object
```

Reply:
Not sure what I was thinking when I wrote this.... :) sorry about that, ignore me ;) Thanks for flagging